compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -659,18 +659,28 @@ fn infer_placeholder_type(
                 format!("{}: {}", item_ident, ty),
                 Applicability::MachineApplicable,
             )
-            .emit();
+            .emit_unless(matches!(ty.kind(), ty::Error(_)) || ty.references_error());
```

Comment:
```suggestion
            .emit_unless(ty.references_error());
```

If `let ty::Error(_) = ty.kind()`, then `ty.references_error()` would also be true, I think.